workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/RequireAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Improbable.Gdk.Core
+{
+    /// <returns>
+    ///     Marks fields/properties of MonoBehaviours that require a Reader or Writer to be injected into them.
+    /// </returns>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
```

Comment:
Do we need to be able to inject properties at all? If we only do fields then users can still expose these outside of the class. I don't think I've ever seen this done with a property. It also get's us into the world of trying to set values to properties that might do something odd with the `value` field. This will likely always work for autoproerties, but is breakable elsewhere.

Reply:
Since it was supported in the SDK like this, I assumed it should be brought over to make it as easy as possible to move SDK code to the GDK.

Reply:
Let's avoid breaking changes for now and think about modifications later.

Reply:
How well did the sdk support it though? If I created a property that isn't an auto property what would happen?

Reply:
It only makes it easier if properties were used as injection targets. Why not start with the simplest baseline (fields) and then add to it later, if it's absolutely necessary.

Reply:
Looking through the SDK implementation and the C# reflection documentation, I'm pretty sure it would call your custom setter to inject the reader/writer, and this is what this code will end up doing as well.